mx.c

```diff
@@ -155,7 +155,28 @@ const struct MxOps *mx_get_ops(enum MailboxType type)
  */
 static bool mutt_is_spool(const char *str)
 {
-  return mutt_str_strcmp(C_Spoolfile, str) == 0;
+  struct Url *ua = url_parse(str);
+  struct Url *ub = url_parse(C_Spoolfile);
+
+  if (!ua || !ub)
+    return false;
+
+  bool is_spool = false;
+  if (ua->scheme != ub->scheme)
+    goto done;
+  if (mutt_str_strcasecmp(ua->host, ub->host) != 0)
+    goto done;
+  if (mutt_str_strcasecmp(ua->path, ub->path) != 0)
+    goto done;
+  if (ua->user && ub->user && mutt_str_strcmp(ua->user, ub->user) != 0)
+    goto done;
+
+  is_spool = true;
```

Comment:
couldn't you simplify by having something such as:
`is_spool = ua->scheme == ub->scheme && mutt_str_strcasecmp(ua->host, ub->host) == 0 && mutt_str_strcasecmp(ua->path, ub->path) == 0 && mutt_str_strcmp(ua->user, ub->user) == 0;` ?
It woudl avoid the goto and mutt_str_strcmp handles null pointer

Reply:
I could put scheme, host, and path together, but I only want to compare usernames if they're both non-null.

Reply:
fair enough

Reply:
Like this: 
```c
  const bool is_spool =                                                             
    (ua->scheme == ub->scheme) &&                                                   
    (mutt_str_strcasecmp(ua->host, ub->host) == 0) &&                               
    (mutt_str_strcasecmp(ua->path, ub->path) == 0) &&                               
    (!ua->user || !ub->user || (mutt_str_strcmp(ua->user, ub->user) == 0));
```

Reply:
I personally find it clearer :)